### PR TITLE
Another attempt at switching from Mail::Sender to Email::Sender.  

### PR DESCRIPTION
**Pull request by pstaabp**
A few more fixes for the switch. 

JITAR sets have an email option.  To test this 

1. create a new set of type "Just In Time" 
2. Save the set and you will see the JITAR options. Make sure the email option is set to YES.
3. Give it at least 2 problems. Click "Expand All" and slide the 2nd problem to the right. 
4. On the second problem (labelled 1.1), change the attempts to 1.  and counts for parent to YES
5. On the first problem, set max attempts to 1 and Att. to open children to `max`.  
6. Go to the problem set as a student and answer the questions wrong.  You should get an email about the homework set. 

I also added an option to change the port in site.conf.dist and explained a bit more. Also in this, since the tls_allowed flag is only a mail option, I changed this to `$mail{tls_allowed}`. I know this will require people updating site.conf, so let me know if this is a bad idea.  

If anyone has a way to check that the port option is working.  I don't currently.

### CODE REVIEW

File: lib/WeBWorK/ContentGenerator/ProblemUtil/ProblemUtil.pm [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ################################################################################
 # WeBWorK Online Homework Delivery System
-# Copyright � 2000-2007 The WeBWorK Project, http://openwebwork.sf.net/
+# Copyright � 2000-2007 The WeBWorK Project, http://openwebwork.sf.net/
```
Comment (Alex-Jordan) [inline lib/WeBWorK/ContentGenerator/ProblemUtil/ProblemUtil.pm]: Another encoding issue here with the copyright symbol.

**Comment by Alex-Jordan** [comment]
> I know this will require people updating site.conf, so let me know if this is a bad idea.

I had to update it anyway. The `site.conf` I had did not have anything about `$tls_allowed`, although I'm not sure how old my `site.conf` was.

**Comment by Alex-Jordan** [comment]
After switching to `develop` to begin testing this, problems are now giving me this error:

```
    Failed to evaluate module Rserve: Can't locate Rserve.pm in @INC (you may need to install the Rserve module) (@INC contains: /opt/webwork/pg/lib /opt/webwork/webwork2/lib /etc/perl /usr/local/lib/perl/5.18.2 /usr/local/share/perl/5.18.2 /usr/lib/perl5 /usr/share/perl5 /usr/lib/perl/5.18 /usr/share/perl/5.18 /usr/local/lib/site_perl . /etc/apache2) at (eval 749) line 1.
    Failed to evaluate module Class::Tiny: Can't locate Rserve.pm in @INC (you may need to install the Rserve module) (@INC contains: /opt/webwork/pg/lib /opt/webwork/webwork2/lib /etc/perl /usr/local/lib/perl/5.18.2 /usr/local/share/perl/5.18.2 /usr/lib/perl5 /usr/share/perl5 /usr/lib/perl/5.18 /usr/share/perl/5.18 /usr/local/lib/site_perl . /etc/apache2) at (eval 749) line 1.
    Failed to evaluate module IO::Handle: Can't locate Rserve.pm in @INC (you may need to install the Rserve module) (@INC contains: /opt/webwork/pg/lib /opt/webwork/webwork2/lib /etc/perl /usr/local/lib/perl/5.18.2 /usr/local/share/perl/5.18.2 /usr/lib/perl5 /usr/share/perl5 /usr/lib/perl/5.18 /usr/share/perl/5.18 /usr/local/lib/site_perl . /etc/apache2) at (eval 749) line 1.
```

I tried to install "Rserve" using MCPAN, but that is unrecognized. I also installed "Statistics::R::IO::Rserve" on a hunch, but that has not helped. Any ideas what package I need?

**Comment by Alex-Jordan** [comment]
Short version: Peter's test worked for me.

My issue with `Rserve` mentioned in this thread was that I was not on the develop branch of pg. 

I went through Peter's testing steps. When I got to step 6 and submitted a wrong answer as a student, I got a message about the course's data tables needing to be upgraded. So I upgraded them.

Then when I submitted a wrong answer, it seemed fine but no email came. Then I realized I again needed to upgrade `site.conf` to change `$tls_allowed` to `$mail{tls_allowed}`. After this (and setting `$mail{tls_allowed}` to `0` given my smtp server) I received the email notification I was supposed to. (But still with flags from Google as noted in #771.

**Comment by pstaabp** [comment]
@Alex-Jordan In general, if you're testing anything off develop (either pg or webwork2), you'll need to be on the develop branch of the other. 

My SMTP server does not allow the TLS security setting, so I can't test that either. I think @mgage can.  
 I think the google problem maybe their extra layer of security.  I am working with our IT dept right now to get the SSL/TLS working.  I'm surprised it isn't set up, but it maybe not set for situations like webwork which connects in the way it does.

**Comment by mgage** [comment]
When I set the port number manually I get this error message:

Error messages

port is a read-only accessor at /opt/webwork/webwork2/lib/WeBWorK/ContentGenerator/Feedback.pm line 253.


I think you have to set the port number when the transport is created.

**Comment by mgage** [comment]
I have a fix for the bug where the port can't be set.  It also adds more error messages to help diagnose mismatches and factors out the creation of the transport layer to a function in 
WeBWorK::ContentGenerator. I will issue a PR targeted at ww2/sendmail

**Comment by mgage** [comment]
JITAR emailing behaves properly.

**Comment by pstaabp** [comment]
I think this looks good now.   Especially, Mike, if you have been able to test under SSL conditions.  We will need to document this change in the release notes.

**Comment by mgage** [comment]
Thanks Peter.  I'll pull it now.

**Pull request merged**